Explorer/Assets/DCL/MapRenderer/MapLayers/Atlas/SatelliteAtlas/SatelliteChunkController.cs

```diff
@@ -64,10 +74,17 @@ public async UniTask LoadImageAsync(Vector2Int chunkId, float chunkWorldSize, Ca
             atlasChunk.MainSpriteRenderer.color = INITIAL_COLOR;
             var url = $"{CHUNKS_API}{chunkId.x}%2C{chunkId.y}.jpg";
 
-            Texture2D texture = (await webRequestController.GetTextureAsync(new CommonArguments(URLAddress.FromString(url)),
-                new GetTextureArguments(false), GetTextureWebRequest.CreateTexture(TextureWrapMode.Clamp, FilterMode.Trilinear)
-                                                                    .SuppressExceptionsWithFallback(Texture2D.whiteTexture, reportContext: ReportCategory.UI),
-                linkedCts.Token, ReportCategory.UI))!;
+            //TODO Potential lick due CacheCleaner behaviour is unclean
+            OwnedTexture2D ownedTexture = (

```

Comment:
You should release whatever is owned natively apart from the managed texture itself, otherwise there is indeed a leak.
It's not connected to caching because here `Satellite` pictures are referenced and never released